packages/core/js-client/src/services/Sig.ts

```diff
@@ -73,10 +73,10 @@ export class Sig {
   /**
    * Signs the data using key pair's private key. Required by aqua
    */
-  async sign(
-    data: number[],
-    context: ParticleContext,
-  ): Promise<SignReturnType> {
+  async sign({
+    args: [data],
+    context,
+  }: MethodArgs<[number[]]>): Promise<SignReturnType> {
```

Comment:
I think that's pretty nice to work with now, thanks. I would only suggest to rename this to maybe `ServiceFnArgs` since these are supposed to be built-in service functions